pkg/model/keycloak_deployment.go

```diff
@@ -245,9 +245,39 @@ func KeycloakDeployment(cr *v1alpha1.Keycloak, dbSecret *v1.Secret) *v13.Statefu
 	} else if cr.Spec.MultiAvailablityZones.Enabled {
 		keycloakStatefulset.Spec.Template.Spec.Affinity = KeycloakPodAffinity(cr)
 	}
+
+	setSSLDBConnection(&keycloakStatefulset.Spec.Template.Spec, dbSecret)
+
 	return keycloakStatefulset
 }
 
+// In case SSLMODE property is not nil, we'll configure an env variable and a mount to the CRT secret
+// to connect to the DB using SSL
+func setSSLDBConnection(podSpec *v1.PodSpec, dbSecret *v1.Secret) {
+	if dbSecret != nil && dbSecret.Data[DatabaseSecretSslModeProperty] != nil {
+		podSpec.Containers[0].Env = append(podSpec.Containers[0].Env, v1.EnvVar{
+			Name:  DatabaseConnectionParamsProperty,
+			Value: "ssl=true;sslMode=" + string(dbSecret.Data[DatabaseSecretSslModeProperty]),
+		})
+
+		// We also expect that the secret with the CRT is present in the namespace
+		podSpec.Volumes = append(podSpec.Volumes, v1.Volume{
+			Name: DatabaseSecretSslCert + "_vol",
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: DatabaseSecretSslCert,
+					Optional:   &[]bool{true}[0],
+				},
+			},
+		})
+		podSpec.Containers[0].VolumeMounts = append(podSpec.Containers[0].VolumeMounts, v1.VolumeMount{
+			Name:      DatabaseSecretSslCert + "_vol",
+			ReadOnly:  true,
+			MountPath: "/home/jboss/.postgresql",
+		})
```

Comment:
We do not need to always mount the certificate. E.g. in case of [`sslMode=allow`](https://jdbc.postgresql.org/documentation/head/ssl-client.html) the cert is not verified and therefore it doesn't need to be mounted.

I'd suggest we mount it only if the secret with the cert is present in the cluster.

Reply:
Very good point :+1:

Reply:
Considering the point you opened..... the same that applies to "allow" would apply to "disable" ? any special treatment should we do to "prefer" ? 
<p class="c3" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 1rem; color: rgb(51, 103, 145); font-family: &quot;Open Sans&quot;, sans-serif; font-size: 14.4px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 600; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Table 31-1. SSL Mode Descriptions</p>

sslmode | Eavesdropping protection | MITM protection | Statement
-- | -- | -- | --
disable | No | No | I don't care about security, and I don't want to pay the overhead of encryption.
allow | Maybe | No | I don't care about security, but I will pay the overhead of encryption if the server insists on it.
prefer | Maybe | No | I don't care about encryption, but I wish to pay the overhead of encryption if the server supports it.
require | Yes | No | I want my data to be encrypted, and I accept the overhead. I trust that the network will make sure I always connect to the server I want.
verify-ca | Yes | Depends on CA-policy | I want my data encrypted, and I accept the overhead. I want to be sure that I connect to a server that I trust.
verify-full | Yes | Yes | I want my data encrypted, and I accept the overhead. I want to be sure that I connect to a server I trust, and that it's the one I specify.

Reply:
I would not implement any special behavior for the specific SSL modes in the operator. IMHO the operator doesn't need to understand the modes, doesn't need to care about their meaning – it just needs to pass it to the correct env var in the image.

I think we could keep it simple: if `sslMode` is specified AND the secret with the cert is found, we mount it. Don't need to based on the modes.

WDYT?

Reply:
:+1: